src/main/java/io/pravega/connectors/flink/FlinkPravegaReader.java

```diff
@@ -165,6 +180,55 @@ void initialize() {
         //       See https://github.com/pravega/flink-connectors/issues/130.
         log.info("Creating reader group: {}/{} for the Flink job", this.readerGroupScope, this.readerGroupName);
         createReaderGroup();
+        if (isEventTimeMode()) {
+            Preconditions.checkArgument(readerGroup.getStreamNames().size() == 1,
+                    "Only 1 Pravega stream is allowed in the event-time mode");
+        }
+    }
+
+    private boolean isEventTimeMode() {
+        return assignerWithTimeWindows != null;
```

Comment:
It's unfortunate that there is no way to get the actual Time Characteristic of the environment.  The connector is making an ASSUMPTION that if the user doesn't specify an assigner then we're working in `EventTime` but it could be that the environment is working in `EventTime` but that they forgot to add the assigner.

It would be nice if there was a way to make this assertion that the connector and the stream are setup to work in the same time characteristic.

Reply:
There should be choice that user uses their own watermark by `assignTimestampAndWatermarks` operator and do not use Pravega watermarks. There is also a circumstance that the writer of the `stream` did not call `noteTime()`, we can't block the reader to use `EventTime` mode to read the stream. I think "no assertion" is reasonable here.